infra/app/service/main.tf

```diff
@@ -66,9 +69,25 @@ data "aws_rds_cluster" "db_cluster" {
   cluster_identifier = local.database_config.cluster_name
 }
 
-data "aws_iam_policy" "db_access_policy" {
+resource "aws_iam_policy" "db_app_access_policy" {
+  count  = module.app_config.has_database ? 1 : 0
+  name   = local.database_config.app_access_policy_name
+  policy = data.aws_iam_policy_document.db_app_access_policy[0].json
+}
```

Comment:
why are we creating the policy here? it should already have been created by the database module and accessed through the data source